lib/inferno/dsl/fhir_client.rb

```diff
@@ -60,26 +60,50 @@ def fhir_clients
         @fhir_clients ||= {}
       end
 
+      # Move parameters from body to the path for GET operations
+      #
+      # @param path [String]
+      # @param body [FHIR::Parameters] Must all be primitive if making GET request
+      # @private
+      def body_to_path(path, body)
+        body.parameter.reduce("#{path}?") do |new_path, x|
+          valid = x.valid? && x.part.empty? && x.resource.nil? # Parameter is valid
+          param_val = x.to_hash.except('name') # should contain only one value if is a valid parameter, checked above
+          if valid && !param_val.empty? && FHIR.primitive?(datatype: param_val.keys[0][5..], value: param_val.values[0])
+            "#{new_path}#{x.name}=#{param_val.values[0]}&"
```

Comment:
Rather than reducing to a string, you should be able to reduce to a hash of name/value pairs, and use an existing method to convert the hash into a query string.

Reply:
Updated to reduce to a hash, then use Hash.to_query as advised above.